src/Tasks/Microsoft.NET.Build.Tasks/ResolvePackageAssets.cs

```diff
@@ -570,7 +571,7 @@ internal ITaskItem[] ReadItemGroup()
 
             private ITaskItem ReadItem()
             {
-                var item = new TaskItem(_reader.ReadString());
+                var item = new TaskItem(ProjectCollection.Escape(_reader.ReadString()));
                 int metadataCount = _reader.ReadInt32();
```

Comment:
Can we do this on write instead of read? That would prevent doing the string scanning on every incremental build.

Reply:
Even better, perhaps. Looks like there's an `EvaluatedIncludeEscaped` setter. So we could default construct and set that. For some reason, it's an explicit implementation of ITaskItem2 so annoying to call.

Reply:
Nevermind, the setter does the same thing as ItemSpec setter. It's the ItemSpec getter that unescapes. Sigh.

Reply:
Good idea. I'll make that change.